File: 2 КУРС/УЧ/1178/ConsoleApplication1.cs
using System;

// ConsoleApplication1.cpp : Этот файл содержит функцию "main". Здесь начинается и заканчивается выполнение программы.
//


public class City
{
	public int id;
	public int x;
	public int y;

//C++ TO C# CONVERTER WARNING: 'const' methods are not available in C#:
//ORIGINAL LINE: bool operator <(const City& other) const
	public static bool operator < (City ImpliedObject, City other)
	{
		if (ImpliedObject.x < other.x)
		{
			return true;
		}
		else
		{
			if (ImpliedObject.x == other.x)
			{
				return ImpliedObject.y < other.y;
			}
			else
			{
				return false;
			}
		}
	}

//C++ TO C# CONVERTER WARNING: 'const' methods are not available in C#:
//ORIGINAL LINE: int dist() const
	public int dist()
	{
		return x * x + y * y;
	}

	public void print()
	{
		Console.Write(id);
		Console.Write("(");
		Console.Write(x);
		Console.Write(", ");
		Console.Write(y);
		Console.Write("): ");
		Console.Write(dist());
		Console.Write("\n");
	}
}

File: 2 КУРС/УЧ/1178/zadanie_1178/ConsoleApplication1/ConsoleApplication1.cpp
// ConsoleApplication1.cpp : Этот файл содержит функцию "main". Здесь начинается и заканчивается выполнение программы.
//

#include <iostream>
#include <stdio.h>
#include <vector>
#include <algorithm>
using namespace std;

struct City
{
public:
	int id;
	int x;
	int y;

	bool operator <(const City& other) const
	{
		if (x < other.x)
			return true;
		else
			if (x == other.x)
				return y < other.y;
			else
				return false;
	}

	int dist() const
	{
		return x * x + y * y;
	}

	void print()
	{
		cout << id << "(" << x << ", " << y << "): " << dist() << endl;
	}
};

int main()
{
	int count;
	cin >> count;
	std::vector<City> v(count);
	for (int i = 0; i < count; ++i)
	{
		int x, y;
		cin >> x >> y;
		v[i].id = i + 1;
		v[i].x = x;
		v[i].y = y;
	}
	sort(v.begin(), v.end());
	for (int i = 0; i < count / 2; ++i)
		cout << v[i * 2].id << ' ' << v[i * 2 + 1].id << endl;
}

